src/imio/esign/browser/settings.py

```
@@ -38,6 +38,15 @@ def validate_vat_number(va_nb):
 
     return True
 
+def validate_signing_users_email_content(value):
+    """Allow only known placeholders in the email template."""
+    allowed = {"fullname", "firstname", "lastname", "email", "userid"}
+    for _, field, _, _ in Formatter().parse(value or ""):
```

Ruff (0.14.14)

[error] Import `_` from line 3 shadowed by loop variable

(F402)